ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetPayloadBodiesByRangeV1.java

```diff
@@ -0,0 +1,85 @@
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
+
+import static org.hyperledger.besu.util.Slf4jLambdaHelper.traceLambda;
+
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.BlockResultFactory;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EngineGetPayloadBodiesByRangeV1 extends ExecutionEngineJsonRpcMethod {
+  private static final Logger LOG = LoggerFactory.getLogger(EngineGetPayloadBodiesByRangeV1.class);
+  private final BlockResultFactory blockResultFactory;
+
+  public EngineGetPayloadBodiesByRangeV1(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final BlockResultFactory blockResultFactory,
+      final EngineCallListener engineCallListener) {
+    super(vertx, protocolContext, engineCallListener);
+    this.blockResultFactory = blockResultFactory;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ENGINE_GET_PAYLOAD_BODIES_BY_RANGE_V1.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext request) {
+    engineCallListener.executionEngineCalled();
+
+    final long startBlockNumber = request.getRequiredParameter(0, Long.class);
+    final long count = request.getRequiredParameter(1, Long.class);
+
+    traceLambda(
+        LOG,
+        "{} parameters: start block number {} count {}",
+        () -> getName(),
+        () -> startBlockNumber,
+        () -> count);
+
+    final Blockchain blockchain = protocolContext.getBlockchain();
+
+    final long latestKnownBlockNumber = blockchain.getChainHeadBlockNumber();
+    final long upperBound = startBlockNumber + count;
+    final long finalBlockNumber =
+        latestKnownBlockNumber < upperBound ? latestKnownBlockNumber + 1 : upperBound;
+
+    return new JsonRpcSuccessResponse(
+        request.getRequest().getId(),
+        blockResultFactory.payloadBodiesCompleteV1(
+            LongStream.range(startBlockNumber, finalBlockNumber)
+                .mapToObj(
+                    blockNumber ->
+                        blockchain
+                            .getBlockHashByNumber(blockNumber)
+                            .flatMap(blockchain::getBlockBody))
```

Comment:
@matkt can/should this be optimised?

Reply:
We spoke and we cannot save a db call since we don't key the database by blockNumber. Also, deserializing txs when we retrieve the blockBody is expensive, but required by the API in this case.

Reply:
There was some discussion around introducing limits for these requests (github) when the methods were been scoped. Maybe we should consider that as well?